src/core/EventDispatcher.js

```diff
@@ -4,39 +4,10 @@
 
 THREE.EventDispatcher = function () {};
 
-//
-// [Deprecation]
-//
+Object.assign( THREE.EventDispatcher.prototype, {
 
-THREE.EventDispatcher.prototype = Object.assign( Object.create( {
-
-	constructor: THREE.EventDispatcher,
-
-	apply: function ( object ) {
-
-		console.warn( "THREE.EventDispatcher: .apply is deprecated, " +
-				"just inherit or Object.assign the prototype to mix-in." );
-
-		object.addEventListener = THREE.EventDispatcher.prototype.addEventListener;
-		object.hasEventListener = THREE.EventDispatcher.prototype.hasEventListener;
-		object.removeEventListener = THREE.EventDispatcher.prototype.removeEventListener;
-		object.dispatchEvent = THREE.EventDispatcher.prototype.dispatchEvent;
-
-	}
-
-	// Notes:
-	// - The prototype chain ensures that Object.assign will not copy the
-	//   properties within this block.
-	// - When .constructor is not explicitly set, it is not copied either,
-	//   so use the disabled code below so doesn't need to be clobbered.
-
-} ), {
-
-//
-// [/Deprecation]
-//
-
-//Object.assign( THREE.EventDispatcher.prototype, {
+	// Note: *No* constructor here, this would clobber that property
```

Comment:
When using `Object.assign( MyBareCtor.prototype, { ... } )` no.

When inheriting from another class, `Object.create( MyBaseCtor.prototype )` will have the wrong constructor (that of the base class) and should normally be overwritten with the derived constructor.

For classes that are intended to be used as *mixins* like `EventDispatcher` in Three.js (a second, "not really abstract base", but an *aspect* taken from somewhere), the most derived prototype must not have an own enumerable constructor - else it will clobber the constructor on the prototype being mixed into.